src/lib.rs

```diff
@@ -308,22 +354,37 @@ impl<T: AsRef<[u8]>> Base32Len for T {
 
 /// A trait to convert between u8 arrays and u5 arrays without changing the content of the elements,
 /// but checking that they are in range.
-pub trait CheckBase32<T: AsRef<[u5]>> {
+pub trait CheckBase32<T> {
     /// Error type if conversion fails
     type Err;
 
     /// Check if all values are in range and return array-like struct of `u5` values
     fn check_base32(self) -> Result<T, Self::Err>;
 }
 
-impl<'f, T: AsRef<[u8]>> CheckBase32<Vec<u5>> for T {
+impl<'f, T, U: AsRef<[u8]>> CheckBase32<T> for U
+where
+    T: AsRef<[u5]>,
+    T: core::iter::FromIterator<u5>,
+{
     type Err = Error;
 
-    fn check_base32(self) -> Result<Vec<u5>, Self::Err> {
+    fn check_base32(self) -> Result<T, Self::Err> {
         self.as_ref()
             .iter()
             .map(|x| u5::try_from_u8(*x))
-            .collect::<Result<Vec<u5>, Error>>()
+            .collect::<Result<T, Error>>()
+    }
+}
+
+impl<'f, U: AsRef<[u8]>> CheckBase32<()> for U {
+    type Err = Error;
+
+    fn check_base32(self) -> Result<(), Error> {
+        self.as_ref()
+            .iter()
+            .map(|x| u5::try_from_u8(*x).map(|_| ()))
+            .collect::<Result<(), Error>>()
```

Comment:
This should do the same imo:

```suggestion
            .find(|r| r.is_err())
            .unwrap_or(Ok(()))
```

Reply:
Oh, awesome it's that simple!